pkg/operator/deploy/helpers.go

```diff
@@ -45,47 +46,171 @@ func decodeYAMLManifestToObject(path string, resource interface{}) error {
 	})
 	crds = append(crds, CRD{
 		Name: "reportdatasources.metering.openshift.io",
-		Path: filepath.Join(deploy.ansibleOperatorManifestsLocation, reportdatasourceFile),
+		Path: filepath.Join(manifestDir, pathToCRDMap["reportDataSource"]),
 		CRD:  new(apiextv1beta1.CustomResourceDefinition),
 	})
 	crds = append(crds, CRD{
 		Name: "meteringconfigs.metering.openshift.io",
-		Path: filepath.Join(deploy.ansibleOperatorManifestsLocation, meteringconfigFile),
+		Path: filepath.Join(manifestDir, pathToCRDMap["meteringConfig"]),
 		CRD:  new(apiextv1beta1.CustomResourceDefinition),
 	})
 
-	deploy.crds = crds
+	return crds
+}
+
+func getMeteringAnsiblePath(manifestDir, platform string) (string, error) {
+	if manifestDir == "" {
+		return "", fmt.Errorf("Failed to set the $DEPLOY_MANIFESTS_DIR or --deploy-manifests-dir flag to a non-empty value")
+	}
+
+	deployDir, err := filepath.Abs(manifestDir)
+	if err != nil {
+		return "", fmt.Errorf("Failed to get the absolute path of the manifest/deploy directory %s: %v", manifestDir, err)
+	}
+
+	dirStat, err := os.Stat(manifestDir)
+	if os.IsNotExist(err) {
+		return "", fmt.Errorf("Failed to get the stat the manifest/deploy directory %s: %v", manifestDir, err)
+	}
+	if !dirStat.IsDir() {
+		return "", fmt.Errorf("Specified deploy directory '%s' is not a directory", manifestDir)
+	}
+
+	var ansibleOperatorManifestDir string
+
+	switch strings.ToLower(platform) {
+	case "upstream":
+		ansibleOperatorManifestDir = filepath.Join(deployDir, upstreamManifestDirname, manifestAnsibleOperator)
+	case "openshift":
+		ansibleOperatorManifestDir = filepath.Join(deployDir, openshiftManifestDirname, manifestAnsibleOperator)
+	case "ocp-testing":
+		ansibleOperatorManifestDir = filepath.Join(deployDir, ocpTestingManifestDirname, manifestAnsibleOperator)
+	default:
+		return "", fmt.Errorf("Failed to set $DEPLOY_PLATFORM or --platform flag to a valid value. Supported platforms: [upstream, openshift, ocp-testing]")
+	}
+
+	dirStat, err = os.Stat(ansibleOperatorManifestDir)
+	if os.IsNotExist(err) {
+		return "", fmt.Errorf("Failed to stat the %s deploy platform directory '%s': %v", platform, ansibleOperatorManifestDir, err)
+	}
+	if !dirStat.IsDir() {
+		return "", fmt.Errorf("Specified %s deploy platform directory '%s' is not a directory", platform, ansibleOperatorManifestDir)
+	}
+
+	return ansibleOperatorManifestDir, nil
+}
+
+// Resource contains information about an individual resource that metering manages
+type Resource struct {
+	IsMeteringConfig bool
+	Path             string
+	Resource         interface{}
+}
+
+// InitObjectFromManifest is the driver function for building up a path to the metering-ansible-operator directory
+// from the base @manifestDir directory and initializing the metering resource/CRD objects from the YAML manifests.
+func InitObjectFromManifest(manifestDir, meteringConfigCRFile string, cfg *Config, resources *MeteringResources) ([]CRD, error) {
```

Comment:
I find the way we're handling different objects so inconsistently a bit odd. We have both `cfg` and `resources`, when we can probably have one. We return CRDs, when it could probably be part of MeteringResources.

How about:

- We return `resources *MeteringResources` instead of taking it as an argument, and instead of returning CRDs.
- CRDs is a field on `*MeteringResources`
- MeteringConfig is a field on `*MeteringResources`

Reply:
I tried going the route of eliminating the need for the `deployer.crds` field, but wasn't able to find an elegant solution that didn't involve using pointers and type assertion.

Previously I tried doing something similar to this:
```go
	meteringResourceMap := map[string]Resource{
		"hivetables.metering.io": Resource{
			Path:             hiveTablePath,
                        IsCRD:          true,
			Resource:     new(apiextv1beta.CustomResourceDefinition),
		},
        ...
```

If you have any idea on how to consolidate these resources into one grouping, then I don't think it would be that hard to adapt the metering CRD install/uninstall deploy methods to check for this boolean field.

Reply:
Unless I misunderstood, and what you meant was to adapt the `MeteringResources` structure to include managing the slice of CRDs (and `MeteringConfig`), which shouldn't be that difficult to do either.

Reply:
I'm not suggesting changing the logic, just moving it. So the latter comment.